Chart Legends docs (#1888)
telerik/blazor-docs: components/chart/legend.md

---
title: Legend
page_title: Chart Legend

> description: Discover the possibilities of the Legend for the Blazor Chart. Learn how to show it, and explore the customization options.

Reviewer: ```suggestion
description: Discover the capabilities of the Chart Legend in Telerik UI for Blazor. Learn how to show it, and explore the customization options.
```

slug: chart-legend
tags: telerik,blazor,chart,legend,customizations
published: True
position: 100
---

# Telerik Chart Legend

> The Telerik Chart for Blazor has the capability to showcase a visual guide offering details about the elements depicted in the chart. This article explores how to add a Chart legend, identify its building blocks, and customizing the legend.

Reviewer: ```suggestion
The Telerik Chart for Blazor can show a visual guide with details about the series or elements in the Chart. This article explores how to add a Chart legend, identify its building blocks, and customize the legend appearance.
```

## Adding a Legend

1. Add the `<ChartLegend>` child tag and set the `Visible` parameter to `true`
1. Add the `Name` parameter to all Chart series that must be visible in the legend.

> ## Chart Legend Customization

Reviewer 2: Also, would Styling be a better term here?

> You can customize the legend of the Telerik Chart for Blazor by adding nested (child) tags to the `<ChartLegend>` and use their parameters for fine-tuning. 

Reviewer: No need to use the full component name, except in 1-2 places at the top.

```suggestion
You can customize the Chart legend by adding nested (child) tags to the `<ChartLegend>` and use their parameters for fine tuning. 
```

The structure of the nested tags is `<ChartLegend*Specifics*>`, where the specifics can be:

* `Title`
* `Item`
* `Border`

>note Use the IntelliSense to explore the nested tags.

> ## Legend Settings in the Chart Series

Reviewer 2: Consider `## Customizing Individual Items in the Chart Legend` instead. Also would Styling fit better here?

> ## Chart Legend Series Customization

Reviewer: This heading doesn't make it clear what is the relationship between the series and the legend. Consider something like

```suggestion
## Legend Settings in the Chart Series
```

            <ChartSeriesLegendItem Type="@ChartLegendItemType.Area">
                <ChartSeriesLegendItemMarkers Background="#00ff00">
                </ChartSeriesLegendItemMarkers>
            </ChartSeriesLegendItem>
        </ChartSeries>
    </ChartSeriesItems>

    <ChartCategoryAxes>
        <ChartCategoryAxis Categories="@xAxisItems"></ChartCategoryAxis>
    </ChartCategoryAxes>

    <ChartTitle Text="Quarterly revenue"></ChartTitle>

    <ChartLegend Position="ChartLegendPosition.Right" Visible="true">
        <ChartLegendTitle Text="Revenue per product"
                          Background="lightblue"
                          Color="black">
        </ChartLegendTitle>
        <ChartLegendItem>
            <ChartLegendItemMarkers Type="@ChartSeriesMarkersType.Cross"
                                    Background="#00ff00">
            </ChartLegendItemMarkers>
        </ChartLegendItem>
    </ChartLegend>
</TelerikChart>

@code {
    public List<object> series1Data = new List<object>() { 10, 2, 5, 6 };
    public List<object> series2Data = new List<object>() { 5, 8, 2, 7 };
    public string[] xAxisItems = new string[] { "Q1", "Q2", "Q3", "Q4" };
}
````

## See Also

> [Live Chart Legend Customization Demo](https://demos.telerik.com/blazor-ui/chart/legend-customization)

Reviewer 2: Are there relevant links to the API that you can include?
